src/Spago/Command/Fetch.purs

```diff
@@ -276,18 +281,23 @@ run { packages: packagesRequestedToInstall, ensureRanges, isTest, isRepl } = do
       Right _lockfile -> pure dependencies
       Left reason -> writeNewLockfile reason dependencies
 
-type LockfileBuilderResult =
-  { workspacePackages :: Map PackageName Lock.WorkspaceLockPackage
-  , packages :: Map PackageName Lock.LockEntry
-  }
-
-lookupInCache :: forall a k v. Ord k => k -> Ref.Ref (Map k v) -> Spago a (Maybe v)
+lookupInCache :: ∀ a k v. Ord k => k -> Ref.Ref (Map k v) -> Spago a (Maybe v)
 lookupInCache key cacheRef = liftEffect $ Ref.read cacheRef >>= Map.lookup key >>> pure
 
-updateCache :: forall a k v. Ord k => k -> v -> Ref.Ref (Map k v) -> Spago a Unit
+fetchFromCache :: ∀ a k v. Ord k => k -> Ref.Ref (Map k v) -> Spago a v -> Spago a v
```

Comment:
This name gives me the idea that we are just fetching from the cache, but we are also computing the value if it's not there - maybe rename to something like `runWithCache`?

Reply:
I think it's important that we reference that there's a cache involved.

Reply:
Ok, how about `cacheFetchOrCreate`?

Reply:
Maybe just `cached`?

Reply:
`withCache`?

Reply:
Yeah, that works